src/link/Elf.zig

```diff
@@ -1353,14 +1353,20 @@ fn linkWithLLD(self: *Elf, comp: *Compilation) !void {
     // Create an LLD command line and invoke it.
     var argv = std.ArrayList([]const u8).init(self.base.allocator);
     defer argv.deinit();
-    // Even though we're calling LLD as a library it thinks the first argument is its own exe name.
-    try argv.append("lld");
+
+    if (self.base.options.is_native_os and self.base.options.system_linker_hack) {
```

Comment:
I want to veto making --system-linker-hack work for ELF or anything other than `native-macos`

Reply:
I'm good with that.